File: 301/301.cpp
#include <iostream>
using namespace std;

int nim(int a, int b, int c){
	int r;
	r=a^b^c;
	return r;
}

int main(){
	int n;
	int i = 1;
	int zeroes = 0;
	int limit = 1073741824;
	limit++;

	while(i<limit){
		n = nim(i,i*2,i*3);
		if(n==0){
			zeroes++;
		}
		i++;
	}
	cout << zeroes << endl;
}

File: 301/301.py
def nim(a,b,c):
    return a^b^c

limit = 2**30

num_zeroes = 0

i = 1

while i < limit + 1:
    if nim(i, 2*i, 3*i) == 0:
        num_zeroes += 1
    print (i+0.0)/limit
    i += 1

print num_zeroes